profiler/profiler.oql/src/org/netbeans/modules/profiler/oql/engine/api/impl/Snapshot.java

```diff
@@ -166,7 +166,7 @@ public Iterator getClasses() {
         return delegate.getAllClasses().iterator();
     }
 
-    public Iterator getClassNames(String regex)  {
+    public Iterator<JavaClass> getClassNames(String regex)  {
```

Comment:
This is wrong! The iterator defined in line 170 maps the `JavaClass` to its name (see the implementation of `next`) so the Iterator should be `Iterator<String>` and the implementation should be `String next()`. That then correctly lets the compiler tell you something is fishy.

Reply:
Fixed..